### PR TITLE
west.yml: MCUboot synchronization from v3.7-branch

### LINKED ISSUE #80439: [v3.7-branch] Update MCUboot to include fixes
**Describe the bug**
Several fixes (both for code in general and security) have been merged to MCUboot since the release of zephyr 3.7, these fixes (note: fixes only, no new features) up to commit `33de65cebaf78b306501a4195dc0ce4008315e1a` (as of 25th October 2024) should be cherry-picked into a special branch for zephyr 3.7 LTS.

- [x] Branch needs creating in zephyr's MCUboot repository for 3.7 backports
- [x] Commits pushed to branch
- [x] Open PR updating to new commit

### PR DESCRIPTION
**Pull request by nordicjm**
west.yml: MCUboot synchronization from v3.7-branch

Update Zephyr fork of MCUboot to revision:
  ea2410697dd0262edec041a0ccb07fdbde7c1aff

Brings following Zephyr relevant fixes:

  - ea241069 boot: bootutil: boot_record: Fix issue with saving image data
  - 03b64828 boards: mcxn947_qspi: fix mcuboot partition allocation
  - 4a0f42c0 doc: remove repetition
  - 1c47465c boot: zephyr: use EXTRA_CONF_FILE instead of deprecated OVERLAY_CONFIG
  - 8e8ecd91 boot: zephyr: Fix serial recovery for NXP IMX.RT platforms
  - 52c7231a boot: zephyr: Fix Warning 'boot_serial_enter defined but not used'
  - a58d9026 zephyr: hello_world: Fix the main() return type warning
  - 50b7b9ee bootutil: Fix AES and SHA-256 contexts not zeroized with mbedTLS
  - 9b1b95c1 ci: Fix zephyr workflow
  - 809b0e42 bootutil: Add better mode selection checks
  - 0fe7ffd8 ci: Fix FIH Docker image release usage
  - a990c6b6 ci: Update FIH docker to Ubuntu Jammy (22.04)
  - 03d1a443 boot: zephyr: board: Fix nrf54l15pdk Kconfig fragment
  - 323eb8d1 boot: zephyr: MCXN947 currently only does not support swap mode
  - 0a88733d boot_serial: Fix `format` warning
  - 157547c8 boot_serial: Fix `incompatible-pointer-types` warning
  - 60ac682a bootutil: loader: Verify image header before checking image
  - 1cd53982 boot: main: avoid unused build warning
  - 587289d6 bootutil: Fix missing include
  - 610b8e2b bootutil: Fix swap move max app size calculation
  - a4800ce0 imgtool: Add missing encodings to emitter tests
  - 80397e0f imgtool: Fix getpub fails for ed25519 key
  - e29a123d docs: release-notes: Add note on name clash fix
  - a375a14d zephyr: Fix issue with sysbuild if something else is named mcuboot
  - 77b03c7f Fix style issues
  - 35bf48c5 boot: Change boot_enc_load to take slot number instead of image
  - 90836499 docs: release-notes: Add note on fixed zephyr RAM load address
  - 24de0fbc boot: zephyr: Fix RAM load chain load address
  - 197287ce imgtool: Bump cryptography library version
  - 7566edaf boot: Move encryption context invalidation to boot_enc_drop.
  - 86b1ef19 boot: Rename boot_enc_decrypt to boot_decrypt_key
  - 597a1996 boot: boot_serial: change logging to debug level
  - 2fa42bfa doc: readme-zephyr: fix the scratch partition example
  - f2971d20 boot: Add missing boot_enc_init
  - 071b3b8e boot: Remove pointless slot identification
  - 2cd6ce9f sim: Fix MCUBOOT_SWAP_USING_SCRATCH defined in direct-xip and ram-load

Fixes #80439

### CODE REVIEW

**Comment by nordicjm** [comment]
@sylvioalves can you check failing CI here and let me know if there is a commit in MCUboot needed for this to build properly that has been misses?

**Comment by sylvioalves** [comment]
> @sylvioalves can you check failing CI here and let me know if there is a commit in MCUboot needed for this to build properly that has been misses?

@nordicjm checking..

**Comment by sylvioalves** [comment]
In case of esp32c3 error:
```c
/__w/zephyr/bootloader/mcuboot/boot/bootutil/src/bootutil_priv.h:68:2: error: #error "MCUBOOT_DIRECT_XIP_REVERT cannot be enabled unless MCUBOOT_DIRECT_XIP is used"
   68 | #error "MCUBOOT_DIRECT_XIP_REVERT cannot be enabled unless MCUBOOT_DIRECT_XIP is used"
   ```
   
   Is that something you can work out or it is esp32 issue?

**Comment by nordicjm** [comment]
> Edit: The error is the same for all targets as in below.
> 
> ```c
> /__w/zephyr/bootloader/mcuboot/boot/bootutil/src/bootutil_priv.h:68:2: error: #error "MCUBOOT_DIRECT_XIP_REVERT cannot be enabled unless MCUBOOT_DIRECT_XIP is used"
>    68 | #error "MCUBOOT_DIRECT_XIP_REVERT cannot be enabled unless MCUBOOT_DIRECT_XIP is used"
> ```
> 
> Is that something you can work out or it is esp32 (and other socs) that require fixing?

I'll fix that, the esp one is this:
```
ccache /opt/toolchains/zephyr-sdk-0.16.8/xtensa-espressif_esp32s3_zephyr-elf/bin/xtensa-espressif_esp32s3_zephyr-elf-gcc -DBOOTLOADER_BUILD -DESP_PLATFORM -DKERNEL -DK_HEAP_MEM_POOL_SIZE=4096 -DNDEBUG -D__ZEPHYR__=1 -I/__w/zephyr/bootloader/mcuboot/boot/zephyr/include -I/__w/zephyr/bootloader/mcuboot/boot/zephyr/targets -I/__w/zephyr/bootloader/mcuboot/boot/bootutil/include -I/__w/zephyr/bootloader/mcuboot/ext/tinycrypt/lib/include -I/__w/zephyr/zephyr/twister-out/esp32s3_devkitm_esp32s3_procpu/samples/sysbuild/with_mcuboot/sample.sysbuild.with_mcuboot/mcuboot/zephyr/include/generated/zephyr -I/__w/zephyr/zephyr/include -I/__w/zephyr/zephyr/twister-out/esp32s3_devkitm_esp32s3_procpu/samples/sysbuild/with_mcuboot/sample.sysbuild.with_mcuboot/mcuboot/zephyr/include/generated -I/__w/zephyr/zephyr/soc/espressif -I/__w/zephyr/zephyr/soc/espressif/common/include -I/__w/zephyr/zephyr/soc/espressif/esp32s3/. -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/include/bt -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../esp_shared/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../esp_shared/components/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../port/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/efuse/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/efuse/private_include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/efuse/esp32s3/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/efuse/esp32s3/private_include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_adc/deprecated/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_common/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_hw_support/dma -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_hw_support/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_hw_support/include/esp_private -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_hw_support/include/hal -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_hw_support/include/soc -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_hw_support/include/soc/esp32s3 -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_hw_support/port/esp32s3 -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_hw_support/port/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_rom/esp32s3 -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_rom/esp32s3/ld -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_rom/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_rom/include/esp32s3 -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_system/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_system/include/esp_private -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_system/port/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_system/port/include/private -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/hal/esp32s3/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/hal/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/hal/include/hal -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/hal/platform_port/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/log/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/soc/esp32s3/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/soc/esp32s3/ld -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/soc/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/xtensa/esp32s3/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/xtensa/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/xtensa/include/esp_private -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/xtensa/include/xtensa -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_timer/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_timer/private_include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/driver/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/driver/deprecated -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/driver/gpio/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/driver/uart/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/driver/touch_sensor/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/driver/touch_sensor/esp32s3/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/driver/spi/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/spi_flash/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/spi_flash/include/spi_flash -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_pm/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/bootloader_support/bootloader_flash/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/bootloader_support/private_include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/bootloader_support/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/heap/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_psram/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_mm/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_netif/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_coex/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_phy/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_phy/include/esp_private -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_phy/esp32s3/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_wifi/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/esp_event/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/esp_supplicant/src -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/esp_supplicant/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/include/esp_supplicant -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/port/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/src -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/src/crypto -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/src/utils -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/wpa_supplicant/src/eap_peer -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../../components/mbedtls/port/include -I/__w/zephyr/modules/hal/espressif/zephyr/esp32s3/../port/include/boot -I/__w/zephyr/bootloader/mcuboot/boot/bootutil/zephyr/.. -I/__w/zephyr/bootloader/mcuboot/boot/bootutil/zephyr/../include -I/__w/zephyr/bootloader/mcuboot/boot/bootutil/zephyr/../../zephyr/include -I/__w/zephyr/bootloader/mcuboot/boot/bootutil/zephyr/../../../ext/tinycrypt/lib/include -isystem /__w/zephyr/zephyr/lib/libc/minimal/include -isystem /__w/zephyr/zephyr/lib/libc/common/include -fno-strict-aliasing -Werror -Os -imacros /__w/zephyr/zephyr/twister-out/esp32s3_devkitm_esp32s3_procpu/samples/sysbuild/with_mcuboot/sample.sysbuild.with_mcuboot/mcuboot/zephyr/include/generated/zephyr/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always --sysroot=/opt/toolchains/zephyr-sdk-0.16.8/xtensa-espressif_esp32s3_zephyr-elf/xtensa-espressif_esp32s3_zephyr-elf -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/bootloader/mcuboot/boot/zephyr=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -mlongcalls -Wno-unused-variable -Wno-maybe-uninitialized -fstrict-volatile-bitfields -fdump-rtl-expand -std=c99 -MD -MT CMakeFiles/app.dir/main.c.obj -MF CMakeFiles/app.dir/main.c.obj.d -o CMakeFiles/app.dir/main.c.obj -c /__w/zephyr/bootloader/mcuboot/boot/zephyr/main.c
/__w/zephyr/bootloader/mcuboot/boot/zephyr/main.c:52:10: fatal error: esp_image_loader.h: No such file or directory
   52 | #include <esp_image_loader.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
https://github.com/zephyrproject-rtos/zephyr/actions/runs/11515362961/job/32059151909?pr=80425

**Comment by sylvioalves** [comment]
I've update mcuboot with this locally. I can build without any issues. Still checking.

**Comment by nordicjm** [comment]
> I've updated mcuboot with this locally. I can build without any issues. Still checking.
> 
> @nordicjm I did test with build and twister locally. Not a single issue. The only difference is that CI is using sdk 0.16.8 and I am on 0.17.0. Any hints?

Have found the offending commit and removed it

**Review by sylvioalves — DISMISSED** [review]

**Review by butok — DISMISSED** [review]

**Review by de-nordic — DISMISSED** [review]

**Review by aescolar — DISMISSED** [review]

File: submanifests/optional.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ manifest:
       groups:
         - optional
     - name: tf-m-tests
-      revision: d552e4f18b92032bd335d5e3aa312f6acd82a83b
+      revision: pull/11/head
```
Comment (tomi-font) [inline submanifests/optional.yaml:43]: State that it's to 2.1.1 in the commit message. Also, can you add a line about that to the release notes? I can do that first in my PR to main and you can copy from there.

**Review by tomi-font — CHANGES_REQUESTED** [review]

File: submanifests/optional.yaml (hunk 1 above)
Comment (nordicjm) [reply submanifests/optional.yaml:43]: Will do it properly if CI passes, just want to make sure the two are compatible before moving these changes over to the MCUboot repo

**Comment by aescolar** [comment]
Adding the DNM to ensure nobody merges by mistake while we point to "yo" :).
Feel free to remove the label as soon as that is changed.

**Review by tomi-font — CHANGES_REQUESTED** [review]
TF-M side looks good other than the missing release note.

**Comment by nordicjm** [comment]
> TF-M side looks good other than the missing release note.

Release notes are done by release team for LTS releases

**Comment by tomi-font** [comment]
> Release notes are done by release team for LTS releases

:thinking: @nashif @aescolar

**Review by tomi-font — DISMISSED** [review]

**Review by sylvioalves — DISMISSED** [review]

**Comment by nashif** [comment]
This is pulling changes and updated from a different module (tfm) without any documentation of the changes and why they are needed, updates to tf-m should be done on their own and documented, i.e. why they are needed an the bugs they fix, right now this is a 2nd level dependency in a manifest and why tf-m is needed is not obvious.

**Comment by nordicjm** [comment]
@tomi-font can you provide the requested information?

**Comment by tomi-font** [comment]
> @tomi-font can you provide the requested information?

:saluting_face: 

> This is pulling changes and updated from a different module (tfm) without any documentation of the changes

Yeah, this is what I was saying [here](https://github.com/zephyrproject-rtos/zephyr/pull/80425#discussion_r1824537625) and [here](https://github.com/zephyrproject-rtos/zephyr/pull/80425#pullrequestreview-2413455413).

>  and why they are needed, updates to tf-m should be done on their own and documented, i.e. why they are needed an the bugs they fix, right now this is a 2nd level dependency in a manifest and why tf-m is needed is not obvious.

I was originally planning to backport the [TF-M 2.1.1 update](https://github.com/zephyrproject-rtos/zephyr/pull/80667) to 3.7, but it turns out that TF-M and MCUboot versions have an inter-dependency caused by some function signature change.
We already did something for that in main, [here](https://github.com/zephyrproject-rtos/zephyr/pull/77783).
So because of that, I thought the best is to update MCUboot and TF-M simultaneously for 3.7.
The TF-M update is just a regular update from 2.1.0 to 2.1.1 which brings fixes including one for a security vulnerability.

**Comment by nordicjm** [comment]
> > @tomi-font can you provide the requested information?
> 
> 🫡
> 
> > This is pulling changes and updated from a different module (tfm) without any documentation of the changes
> 
> Yeah, this is what I was saying [here](https://github.com/zephyrproject-rtos/zephyr/pull/80425#discussion_r1824537625) and [here](https://github.com/zephyrproject-rtos/zephyr/pull/80425#pullrequestreview-2413455413).
> 
> > and why they are needed, updates to tf-m should be done on their own and documented, i.e. why they are needed an the bugs they fix, right now this is a 2nd level dependency in a manifest and why tf-m is needed is not obvious.
> 
> I was originally planning to backport the [TF-M 2.1.1 update](https://github.com/zephyrproject-rtos/zephyr/pull/80667) to 3.7, but it turns out that TF-M and MCUboot versions have an inter-dependency caused by some function signature change. We already did something for that in main, [here](https://github.com/zephyrproject-rtos/zephyr/pull/77783). So because of that, I thought the best is to update MCUboot and TF-M simultaneously for 3.7. The TF-M update is just a regular update from 2.1.0 to 2.1.1 which brings fixes including one for a security vulnerability.

Could you create a new issue with some details about updating TF-M for 3.7? Then I can link it to the PR

**Comment by tomi-font** [comment]
> Could you create a new issue with some details about updating TF-M for 3.7? Then I can link it to the PR

I mean it's just a normal dependency update as TF-M 2.1 is also an LTS release... No particular bug fix.

**Review by d3zd3z — DISMISSED** [review]

**Comment by nordicjm** [comment]
@nashif can we get this merged for 3.7.1?

**Comment by nashif** [comment]
> @nashif can we get this merged for 3.7.1?

need an entry in the releases notes similar to https://github.com/zephyrproject-rtos/zephyr/blob/v3.7-branch/doc/releases/release-notes-3.7.rst#mbed-tls

**Comment by nordicjm** [comment]
> @nordicjm You can copy [this](https://github.com/zephyrproject-rtos/zephyr/pull/80667/files#diff-ada0effa18bf591a10f0e71db409689e026c8c35cd790a1b0026d25cac005535R556-R560) for TF-M.

Thanks, added

**Review by tomi-font — DISMISSED** [review]
TF-M side looks good, MCUboot release notes are quite verbose, you could possibly just state something generic and e.g. point to this PR, but I'll let the others decide.

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -26,6 +26,59 @@ Mbed TLS was updated to version 3.6.2 (from 3.6.0). The release notes can be fou
...
+  * Revert "boot: Add MCUBOOT_HW_KEY support for image encryption"
+  * boot: Fix ASN.1 for mbedtls >= 3.1
+  * ext: fiat: Use user-defined assert macro
+
+Trusted Firmware-M
+******************
+
+* TF-M was updated to version 2.1.1 (from 2.1.0).
+  The release notes can be found at: https://trustedfirmware-m.readthedocs.io/en/tf-mv2.1.1/releases/2.1.1.html
+
```
Comment (tomi-font) [inline doc/releases/release-notes-3.7.rst:81]:
```suggestion

TF-M 2.1 is an LTS release synchronized with Mbed TLS 3.6 that
will be supported with security and bug fixes until March 2027.
```

Comment (nordicjm) [reply doc/releases/release-notes-3.7.rst:90]: Updated, also changed the lines above to match the mbedtls update at the top

**Review by tomi-font — DISMISSED** [review]
approving TF-M

**Review by nashif — DISMISSED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -26,6 +26,68 @@ Mbed TLS was updated to version 3.6.2 (from 3.6.0). The release notes can be fou
 Mbed TLS 3.6 is an LTS release that will be supported
 with security and bug fixes until at least March 2027.
 
+MCUboot
```
Comment (nashif) [inline doc/releases/release-notes-3.7.rst:29]: this is too verbose of a release note in a module to be included in the zephyr release notes, dont you have something to link to similar to what we do tfm and mbedtls?

Comment (nashif) [reply doc/releases/release-notes-3.7.rst:29]: the issue i see here is that all of those changes might be seen as changes to zephyr and not mcuboot itself, which might be confusing.

Comment (nordicjm) [reply doc/releases/release-notes-3.7.rst:29]: MCUboot has release notes but they're only generated when a release is made, and these are cherry-picked commits so if a release was made, it wouldn't be correct as it's be for mcuboot's `main` not the zephyr-only branch with cherry picked commits

Comment (nordicjm) [reply doc/releases/release-notes-3.7.rst:29]: Can just change to link to this PR instead?

Comment (nordicjm) [reply doc/releases/release-notes-3.7.rst:29]: > You can still make bug fix releases on the branch, no?

It's a zephyr branch of MCUboot, it does not exist upstream. https://github.com/zephyrproject-rtos/mcuboot/tree/v3.7-branch is where it gets cherry picked to

> Either remove them from this PR

Will do

**Review by nashif — APPROVED** [review]

**Pull request merged**